apis/k8ssandra/v1alpha1/cassandraconfig_types.go

```diff
@@ -1041,3 +1049,106 @@ type TrackWarnings struct {
 	LocalReadSize       *int `json:"local_read_size,omitempty"`
 	RowIndexSize        *int `json:"row_index_size,omitempty"`
 }
+
+type ClientEncryptionOptions struct {
+	Enabled bool `json:"enabled"`
+
+	// +optional
+	Optional *bool `json:"optional,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Keystore *string `json:"keystore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	KeystorePassword *string `json:"keystore_password,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Truststore *string `json:"truststore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	TruststorePassword *string `json:"truststore_password,omitempty"`
+
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// +optional
+	AcceptedProtocols *[]string `json:"accepted_protocols,omitempty"`
+
+	// +optional
+	Algorithm *string `json:"algorithm,omitempty"`
+
+	// +optional
```

Comment:
I was thinking we could use an enum here but it seems the supported file types may vary a bit by JVM.

Reply:
yes, I'd stay away from putting constraints here.